lib/elixir/lib/option_parser.ex

```diff
@@ -191,7 +194,7 @@ defmodule OptionParser do
       {[unlock: "path/to/file", unlock: "path/to/another/file"], [], []}
 
   """
-  @spec parse(argv, options) :: {parsed, argv, errors}
+  @spec parse(argv, options) :: {parsed, argv, errors} | no_return
```

Comment:
There is no need for `no_return` here. Most of our uses of `no_return` is wrong because it is expected from Erlang/Elixir that functions can error on non-valid input (case clauses, argument errors, etc).